syntax = "proto3";

package FileServerInternal;

service Server
{
    rpc getCompressedFile(ClientRequest) returns (stream FileInfo){}
    rpc getFileInfoVector(ClientRequest) returns (stream Chunk){}
}

message ClientRequest
{
}

message Chunk
{
        int64          pos             = 1;
        string         path            = 2;
        string         date            = 3;
        int32          buffer          = 4;
        int64          buffer_size     = 5;
}

message FileInfo 
{
        string         path        = 1;
        int64          size        = 2;
        string         date        = 3;
        string         checksum    = 4;
        bool           executable  = 5;
}
